src/vstest.console/Processors/CollectArgumentProcessor.cs

```diff
@@ -172,21 +197,49 @@ internal static void AddDataCollectorToRunSettings(string argument, IRunSettings
                 settings = runSettingsManager.ActiveRunSettings?.SettingsXml;
             }
 
-            var dataCollectionRunSettings = XmlRunSettingsUtilities.GetDataCollectionRunSettings(settings);
-            if (dataCollectionRunSettings == null)
-            {
-                dataCollectionRunSettings = new DataCollectionRunSettings();
-            }
+            var dataCollectionRunSettings = XmlRunSettingsUtilities.GetDataCollectionRunSettings(settings) ?? new DataCollectionRunSettings();
+            var inProcDataCollectionRunSettings = XmlRunSettingsUtilities.GetInProcDataCollectionRunSettings(settings) 
+                ?? new DataCollectionRunSettings(
+                    Constants.InProcDataCollectionRunSettingsName,
+                    Constants.InProcDataCollectorsSettingName,
+                    Constants.InProcDataCollectorSettingName);
 
             // Add data collectors if not already present, enable if already present.
             EnableDataCollectorUsingFriendlyName(argument, dataCollectionRunSettings);
 
             runSettingsManager.UpdateRunSettingsNodeInnerXml(Constants.DataCollectionRunSettingsName, dataCollectionRunSettings.ToXml().InnerXml);
+
+            if (string.Equals(argument, CoverletConstants.CoverletDataCollectorFriendlyName, StringComparison.OrdinalIgnoreCase))
+            {
+                // Add inproc data collector to runsetings if coverlet code coverage is enabled
+                EnableCoverletInProcDataCollector(argument, inProcDataCollectionRunSettings);
+                runSettingsManager.UpdateRunSettingsNodeInnerXml(Constants.InProcDataCollectionRunSettingsName, inProcDataCollectionRunSettings.ToXml().InnerXml);
+            }
         }
 
         internal static void AddDataCollectorFriendlyName(string friendlyName)
         {
             EnabledDataCollectors.Add(friendlyName.ToLower());
         }
+
+        internal static class CoverletConstants
+        {
+            /// <summary>
+            /// Coverlet data collector friendlyname
+            /// </summary>
+            public const string CoverletDataCollectorFriendlyName = "XPlat Code Coverage";
+
+            /// TODO : Finalize Name
+            /// <summary>
+            /// Coverlet data collector assembly qualified name
```

Comment:
`Coverlet in proc data collector assembly qualified name.`
Change in CoverletDataCollectorCodebase as well.